io/fasta/fasta_test.go

```diff
@@ -248,3 +304,86 @@ func TestWrite_error(t *testing.T) {
 	err := Write([]Fasta{}, "/tmp/file")
 	assert.EqualError(t, err, buildErr.Error())
 }
+
+func TestParser(t *testing.T) {
+	parser := NewParser(nil, 256)
+	for testIndex, test := range []struct {
+		content  string
+		expected []Fasta
+	}{
+		{
+			content:  ">humen\nGATTACA\nCATGAT", // EOF-ended Fasta not valid
```

Comment:
we should probably be more permissive and along EOF-ended Fasta as valid?

Reply:
@soypat my latest changes have this entire suite of tests failing on off-by-one errors where there's always one more fasta than what is expected. Any idea what's going on?

Reply:
I'll take a deep dive look today!

Reply:
I'm hesitant to say EOF ended fastas are OK. This means you can't use io.LimitReader to read fastas at all using poly API since its use will return incomplete fastas without warning. The reason this is important to me is because io.LimitReader is a commonly used foundation for writing software resilient to adversarial user input.

Though I'd also like users who want to parse EOF ended fastas to be able to use poly. This is possible since we can return a valid fasta alongside an io.EOF error when the fasta is immediately terminated by an EOF. This will allow users who need this functionality to build their own parsers who treat the EOF case specially.

Note: ParseNext now (correctly?) parses `data/empty.fasta`, returning a single fasta.

Reply:
Maybe we can add a boolean flag to the ParseN, ParseAll, ParseBytes: `acceptEOFTerminatedFastas`?

Reply:
> Maybe we can add a boolean flag to the ParseN, ParseAll, ParseBytes: acceptEOFTerminatedFastas?

This feels a little clunky but may be the way to do it. Should this variadic or explicit?

Reply:
> Note: ParseNext now (correctly?) parses data/empty.fasta, returning a single fasta.

Just made empty.fasta truly empty. It's meant to test the corner cases I just commented on.

Reply:
> This feels a little clunky but may be the way to do it. Should this variadic or explicit?

Maybe we can have parser options? Something like this https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis? Or maybe that's too much- just a struct with fields?